### PR TITLE
[NO ISSUE] draft docs on generating REST APIs

### PR DESCRIPTION
**Pull request by osfameron**

### CODE REVIEW

File: home/modules/contribute/pages/generate-rest-api.adoc [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+== Swagger2Markup
+
+=== Couchbase Analytics
+
+NOTE: TODO, for SimonD
+
+=== Sync Gateway
+
+NOTE: Ian provided the following, which we couldn't get working, as the Swagger2Markup repositories
```
Comment (RayOffiah) [inline home/modules/contribute/pages/generate-rest-api.adoc:188]: Probably shouldn't mention team members by name. And again, this is sort of historical knowledge, so I don't think it needs to be included here.

**Review by RayOffiah — CHANGES_REQUESTED** [review]
Instructions are fine, but I think there's a lot of historical knowledge here that probably belongs in the private wiki rather than the public contribute section. 

And we probably shouldn't mention individual team members.

File: home/modules/contribute/pages/generate-rest-api.adoc [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+When porting the reference docs to Capella, we noticed that the Swagger2Markup toolchain was non-functional,
+and instead used:
+
+. modified, local copies of the yaml specs
+. a modified, local copy of the Core Capella openapi-generator toolchain
+. a custom `[markdown]` tag to parse the content of OpenAPI descriptions.
+
+==== Specs sources
+
```
[the remark this answers is not in the document]
Comment (osfameron) [reply home/modules/contribute/pages/generate-rest-api.adoc:67]: You may be right.. though it's important for now as there's still a reasoning/process behind it
(e.g. we may want to recreate those sources with different criteria)

**Review by RayOffiah — APPROVED** [review]

File: home/modules/contribute/pages/generate-rest-api.adoc [hunk 3]
```diff
@@ -0,0 +1,241 @@
+= Generate REST API from Swagger/OpenAPI
```
Comment (simon-dew) [inline home/modules/contribute/pages/generate-rest-api.adoc:1]: Just a minor point, we (tech comms) don't generate the REST APIs, just their documentation

```suggestion
= Generate REST API Documentation
```

File: home/modules/contribute/pages/generate-rest-api.adoc [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+== Swagger UI
+
+You can include a Swagger UI block using the `swagger_ui::` block macro.
+See for example https://github.com/couchbase/docs-sdk-common/blob/release/7.1/modules/shared/partials/sample-application.adoc
+which is included by all the travel-sample app docs, for example https://docs.couchbase.com/java-sdk/current/hello-world/sample-application.html#rest-api
+
+
+== Swagger2Markup
+
+=== Couchbase Analytics
```
Comment (simon-dew) [inline home/modules/contribute/pages/generate-rest-api.adoc:182]: It's a lot more than just Analytics.

```suggestion
=== Couchbase Server REST APIs
```

**Review by simon-dew — CHANGES_REQUESTED** [review]

File: home/modules/contribute/pages/generate-rest-api.adoc (hunk 4 above)
Comment (simon-dew) [reply home/modules/contribute/pages/generate-rest-api.adoc:182]: @osfameron Here's a brief suggestion to replace the TODO content in this section.

```
Documentation for the Analytics Service REST APIs, Query Service REST APIs, and some Index Service REST APIs is pulled in from AsciiDoc partials which are stored in the `docs` subdirectory of the https://github.com/couchbaselabs/cb-swagger[cb-swagger] repository.

The documentation in that repository is generated automatically, using the swagger2markup gradle plugin, from swagger specification files which are stored in the `src` subdirectory of the same repository.
The swagger2markup gradle plugin also takes additional content from the `src` subdirectory, including examples.

The AsciiDoc partials are built automatically by GitHub Workflows every time a pull request is merged into the https://github.com/couchbaselabs/cb-swagger[cb-swagger] repository.
For further details, refer to the https://github.com/couchbaselabs/cb-swagger#readme[README] in that repository.

Note that documentation for the other Couchbase Server REST APIs is still written by hand.
```

**Review by simon-dew — APPROVED** [review]

**Pull request merged**